Cannot consume CJS build with TypeScript

TypeScript users that don't use ESM are unable to use the include CJS build.

```
error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("fetch-mock")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"`
```

If I patch the `package.json` in `node_modules` and remove the `"type": "module"` field it stops complaining and everything works just fine

This is the minimal code required to reproduce the issue
https://github.com/wolfy1339/fetch-mock-typescript

```
index.ts:1:23 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("fetch-mock")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.

1 import fetchMock from "fetch-mock";
                        ~~~~~~~~~~~~


Found 1 error in index.ts:1
```

--- a/packages/fetch-mock/package.json
+++ b/packages/fetch-mock/package.json
@@ -3,6 +3,7 @@
   "version": "10.1.0",
   "description": "Mock http requests made using fetch (or isomorphic-fetch)",
   "main": "./dist/commonjs.js",
+  "module": "./src/index.js",
   "exports": {
     "types": "./types/index.d.ts",
     "browser": "./src/index.js",
@@ -10,7 +11,6 @@
     "require": "./dist/commonjs.js"
   },
   "types": "./types/index.d.ts",
-  "type": "module",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/wheresrhys/fetch-mock.git",

--- a/packages/fetch-mock/package.json
+++ b/packages/fetch-mock/package.json
@@ -1,6 +1,6 @@
 {
   "name": "fetch-mock",
-  "version": "10.1.0",
+  "version": "10.1.1-alpha.1",
   "description": "Mock http requests made using fetch (or isomorphic-fetch)",
   "main": "./dist/commonjs.js",
   "module": "./src/index.js",

--- a/packages/fetch-mock/package.json
+++ b/packages/fetch-mock/package.json
@@ -1,6 +1,6 @@
 {
   "name": "fetch-mock",
-  "version": "10.1.1-alpha.1",
+  "version": "10.1.0",
   "description": "Mock http requests made using fetch (or isomorphic-fetch)",
   "main": "./dist/commonjs.js",
   "module": "./src/index.js",

--- a/packages/fetch-mock/src/package.json
+++ b/packages/fetch-mock/src/package.json
@@ -1,0 +1,1 @@
+{"type": "module"}

